snippets/csharp/VS_Snippets_CLR_System/system.collections.generic.equalitycomparer/cs/program.cs

```diff
@@ -69,47 +69,33 @@ class BoxSameDimensions : EqualityComparer<Box>
 	public override bool Equals(Box b1, Box b2)
 	{
 		if (b1 == null && b2 == null)
-         return true;
-      else if (b1 == null || b2 == null)
-         return false;
-         
-		if (b1.Height == b2.Height && b1.Length == b2.Length
-							&& b1.Width == b2.Width)
-		{
 			return true;
-		}
-		else
-		{
+		else if (b1 == null || b2 == null)
 			return false;
-		}
+
+		return (b1.Height == b2.Height &&
+				b1.Length == b2.Length &&
+				b1.Width == b2.Width);
```

Comment:
I'm a bit unsure about the indentation here (and similar below for `BoxSameVolume`). It looked correct on my machine, but I also use 4 spaces for tabs (and the file is indented solely using tabs). I could remove one tab from each of the last 2 lines to make the comparison line up, but that will only work in places where tabs = 8 spaces.